File: runme/mcgen/local/datan/exasrc/e3mtx.f
      PROGRAM E3MTX
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      PARAMETER (K=2)
      DIMENSION U(K),W(K)
      DATA U/3.D0,4.D0/
      DATA W/1.D0,1.D0/
C identify program to user
      WRITE(*,*)' Program E3MTX demonstrates use of'
      WRITE(*,*)' MTXGVD,MTXGVT and MTXGVA'
      WRITE(*,*)' for Givens transformations'
      WRITE(*,*)' '
      WRITE(*,*)' Output produced by program E3MTX'
      WRITE(*,'(A)')' ------------------------------------------'
C write initial vectors
      WRITE(*,*)' U ='
      CALL MTXWRT(U,1,K)
      WRITE(*,*)' W ='
      CALL MTXWRT(W,1,K)
      WRITE(*,'(A)')' ------------------------------------------'
      WRITE(*,*)' '
C demonstrate MTXGVD
      CALL MTXGVD(U(1),U(2),C,S)
      WRITE(*,'(A)')' CALL MTXGVD(U(1),U(2),C,S) yields '
      WRITE(*,'(A,F10.5,A,F10.5)')' C = ',C,', S = ',S
      WRITE(*,*)' '
C demonstrate MTXGVT
      CALL MTXGVT(U(1),U(2),C,S)
      WRITE(*,*)' With the above values of C and S'
      WRITE(*,'(A)')' CALL MTXGVT(U(1),U(2),C,S) yields '
      WRITE(*,*)' U ='
      CALL MTXWRT(U,1,K)
      WRITE(*,*)' '
C demonstrate MTXGVA
      CALL MTXGVA(W(1),W(2),C,S)
      WRITE(*,'(A)')' CALL MTXGVA(W(1),W(2),C,S) yields '
      WRITE(*,'(A,F10.5,A,F10.5)')' C = ',C,', S = ',S
      WRITE(*,*)' and W ='
      CALL MTXWRT(W,1,K)
      END

File: runme/mcgen/local/datan/exasrc/e3mtx.c
#include <stdio.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__1 = 1, c__2 = 2;

main(void)
{
    /* Initialized data */
    static double u[2] = { 3.,4. }, w[2] = { 1.,1. }, c, s;

/* identify program to user */
    printf("%s\n", "Program E3MTX demonstrates use of");
    printf("%s\n", "MTXGVD,MTXGVT and MTXGVA");
    printf("%s\n\n", "for Givens transformations.");
    printf("%s\n", "Output produced by program E3MTX.");
    printf("%s\n", "------------------------------------------");
/* write initial vectors */
    printf("%s\n", "U =");
    mtxwrt_(u, &c__1, &c__2);
    printf("%s\n", "W =");
    mtxwrt_(w, &c__1, &c__2);
    printf("%s\n\n", "------------------------------------------");
/* demonstrate MTXGVD */
    mtxgvd_(u, &u[1], &c, &s);
    printf("%s\n", "CALL MTXGVD(U(1),U(2),C,S) yields ");
    printf("%s%10.5lf%s%10.5lf\n\n", "C = ", c, ", S = ", s);
/* demonstrate MTXGVT */
    mtxgvt_(u, &u[1], &c, &s);
    printf("%s\n", "With the above values of C and S");
    printf("%s\n", "CALL MTXGVT(U(1),U(2),C,S) yields ");
    printf("%s\n", "U =");
    mtxwrt_(u, &c__1, &c__2);
    printf("\n");
/* demonstrate MTXGVA */
    mtxgva_(w, &w[1], &c, &s);
    printf("%s\n", "CALL MTXGVA(W(1),W(2),C,S) yields ");
    printf("%s%10.5lf%s%10.5lf\n", "C = ", c, ", S = ", s);
    printf("%s\n", "and W =");
    mtxwrt_(w, &c__1, &c__2);
    return 0;
} /* main */

